src/rust/build.rs

```diff
@@ -0,0 +1,145 @@
+use std::env;
+use std::io::Write;
+use std::path::{Path, MAIN_SEPARATOR};
+use std::process::{Command, Stdio};
+
+fn main() {
+    let target = env::var("TARGET").unwrap();
+    let openssl_static = env::var("OPENSSL_STATIC")
+        .map(|x| x == "1")
+        .unwrap_or(false);
+    if target.contains("apple") && openssl_static {
+        // On (older) OSX we need to link against the clang runtime,
+        // which is hidden in some non-default path.
+        //
+        // More details at https://github.com/alexcrichton/curl-rust/issues/279.
+        if let Some(path) = macos_link_search_path() {
+            println!("cargo:rustc-link-lib=clang_rt.osx");
+            println!("cargo:rustc-link-search={}", path);
+        }
+    }
+
+    let out_dir = env::var("OUT_DIR").unwrap();
+    // FIXME: maybe pyo3-build-config should provide a way to do this?
+    let python = env::var("PYO3_PYTHON").unwrap_or_else(|_| "python3".to_string());
+    println!("cargo:rerun-if-changed=../_cffi_src/");
+    let python_path = match env::var("PYTHONPATH") {
+        Ok(mut val) => {
+            if cfg!(target_os = "windows") {
+                val.push(';');
+            } else {
+                val.push(':');
+            }
+            val.push_str("..");
+            val.push(MAIN_SEPARATOR);
+            val
+        }
+        Err(_) => format!("..{}", MAIN_SEPARATOR),
+    };
+    let output = Command::new(&python)
+        .env("PYTHONPATH", python_path)
+        .env("OUT_DIR", &out_dir)
+        .arg("../_cffi_src/build_openssl.py")
+        .output()
+        .expect("failed to execute build_openssl.py");
+    if !output.status.success() {
+        panic!(
+            "failed to run build_openssl.py, stdout: \n{}\nstderr: \n{}\n",
+            String::from_utf8(output.stdout).unwrap(),
+            String::from_utf8(output.stderr).unwrap()
+        );
+    }
+
+    let python_impl = run_python_script(
+        &python,
+        "import platform; print(platform.python_implementation(), end='')",
+    )
+    .unwrap();
+    println!("cargo:rustc-cfg=python_implementation=\"{}\"", python_impl);
+    let python_include = run_python_script(
+        &python,
+        "import sysconfig; print(sysconfig.get_path('include'), end='')",
+    )
+    .unwrap();
+    let openssl_include =
+        std::env::var_os("DEP_OPENSSL_INCLUDE").expect("unable to find openssl include path");
+    let openssl_c = Path::new(&out_dir).join("_openssl.c");
+
+    let mut build = cc::Build::new();
+    build
+        .file(openssl_c)
+        .include(python_include)
+        .include(openssl_include)
+        .flag_if_supported("-Wconversion")
+        .flag_if_supported("-Wno-error=sign-conversion");
+
+    // Enable abi3 mode if we're not using PyPy.
+    if python_impl != "PyPy" {
+        // cp36
+        build.define("Py_LIMITED_API", "0x030600f0");
```

Comment:
So we have to keep this synced with our wheel builder flags? Presumably there's no way to have it detect the right value like setuptools-rust or maturin do since there's no guarantee this is called with that machinery?

Reply:
Honestly not sure, I don't know where this invocation lives in current code.